polaris-react/src/components/Banner/Banner.scss

```diff
@@ -10,34 +10,29 @@
   }
 
   @if $in-page {
-    border-radius: var(--p-border-radius-2);
+    border-radius: 0;
+    border-width: var(--p-border-width-1) 0;
 
-    @media #{$p-breakpoints-sm-down} {
-      border-radius: 0;
-      border-width: var(--p-border-width-1) 0;
+    #{$se23} & {
+      border: none;
+
+      @include shadow-bevel(
+        $boxShadow: var(--p-shadow-sm),
+        $borderRadius: var(--p-border-radius-0-experimental)
+      );
```

Comment:
Last minute update from pairing with @sarahill. Note: This will need to be applied in `Card` and `LegacyCard`.

<img width="647" alt="image" src="https://github.com/Shopify/polaris/assets/32409546/eea57cad-3e5a-40ad-8aa8-1caee3c69c95">